RIPS/rip-9998.md

```diff
@@ -0,0 +1,306 @@
+
+This amount of consumed gas counts towards the `validationGasLimit` of the transaction, meaning that each
+transaction's validation cost includes `gasUsedByAggregator / transactionCount`.
+This value is referred to as `aggregatedValidationGasUsed`.
+
+### Sender validation frame
+
+The `AA_TX_TYPE_AGGREGATED_SIG` type transaction includes a parameter `transactionCount`.
+This value indicates a number of transactions whose validation has been aggregated.
+
+For AA transactions marked as "aggregated" by the `AA_TX_TYPE_AGGREGATED_SIG` type transaction,
+the validation method `validateTransaction` return value is interpreted as:
+
+```solidity
+
+abi.encodePacked(aggregator, MAGIC_VALUE_SENDER, validBefore, validAfter)
+
+```
+
+The block including this transaction is only valid if the `aggregator` address returned by the `validateTransaction`
+matches the `aggregator` address specified in the `AA_TX_TYPE_AGGREGATED_SIG` transaction.
+
+### RPC methods (eth namespace)
+
+#### `eth_aggregationFeeHistory`
+
+We propose the RPC method `eth_aggregationFeeHistory` to be created.
+This method behaves similarly to `eth_feeHistory` method used to estimate [EIP-1559](./eip-1559.md) gas fees,
+but instead provides history of a per-transaction validation costs,
+`aggregatedValidationGasUsed` and `builderFee`, for a given aggregator.
+The values are calculated as an average across all transactions.
+
+Value of `null` indicates there was no bundle with the given `aggregator` in the given block.
+
+Input: `aggregator address`, `block count`
+
+Returns:
+
+```json
+{
+  "oldestBlock": "100000",
+  "newestBlock": "100003",
+  "aggregatedValidationGasUsed": [
+    "450000",
+    null,
+    "400000",
+    "500000"
+  ],
+  "builderFee": [
+    "0x2386f26fc10000",
+    null,
+    "0x2386f26fc10000",
+    "0x2386f26fc10000"
+  ]
+}
+```
+
+#### `eth_estimateGas`
+
+In case an `aggregator` address is returned by the `validateTransaction` by the `sender`, relies on a
+`eth_aggregationFeeHistory` logic in order to calculate sufficient `validationGasLimit` and `builderFee` values.
+
+Returns `{validationGasLimit, callGasLimit, builderFee}` object.
+
+#### `eth_getTransactionReceipt`
+
+Accepts the Transaction Type `AA_TX_TYPE_AGGREGATED_SIG` hash.
+
+This is a special kind of transaction with `from` field set to the `coinbase` and `to` field set to `aggregator`.
+
+## Rationale
+
+### Fixed transaction order in bundle
+
+The `aggregator` contract maintains control over the number and order of transactions inside the "bundle".
+This means that some of the strategies block builders apply in order to extract MEV from a block become
+harder to do or even impossible.
+
+However, allowing the block builder to arbitrarily reorder transactions in the block, mixing aggregated and regular
+transactions, would make this RIP impractically complex while the MEV extraction was never an intended feature
+in the first place.
```

Comment:
Block builders are free to reorder by splitting aggregated batches (even if they're for the same aggregator).  They need to include multiple AA_TX_TYPE_AGGREGATED_SIG transactions for that aggregator in this case, but can control the order.  Therefore the same strategies are still possible (except if something like `AtomicAggregator` is used to explicitly prevent them).

That's another reason I think the builder should be paying for AA_TX_TYPE_AGGREGATED_SIG transactions rather than splitting their cost among the aggregated transactions.  If the builder chooses to make the batches smaller to control the order, the builder should be the one paying for the additional blockspace and work it causes.